### PR TITLE
Absorbing work from douglashall/learning_sequence

### PR DESCRIPTION
**Pull request by davidjoy**

### CODE REVIEW

File: .env.development [hunk 1]
```diff
@@ -1,7 +1,7 @@
 NODE_ENV='development'
-PORT=8080
+PORT=2000
```
Comment (davidjoy) [inline .env.development:2]: I mean, why not?

File: src/index.jsx [hunk 2]
```diff
@@ -18,7 +19,14 @@ subscribe(APP_READY, () => {
   ReactDOM.render(
     <AppProvider>
       <Header />
-      <ExamplePage />
+      <Switch>
+        <Route
```
Comment (davidjoy) [inline src/index.jsx:23]: This route is a temporary helper for developers to help them find a useful URL.

File: src/learning-sequence/LearningSequencePage.jsx [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+import PropTypes from 'prop-types';
+import { camelCaseObject, getConfig } from '@edx/frontend-platform';
+import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
+import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+
+import PageLoading from './PageLoading';
+
+import messages from './messages';
+
+function useApi(apiFunction, {
```
Comment (davidjoy) [inline src/learning-sequence/LearningSequencePage.jsx:11]: I intend that this `useApi` hook - or something like it - could be moved into frontend-platform/react.

File: src/learning-sequence/LearningSequencePage.jsx [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    failed,
+    error,
+  };
+}
+
+function LearningSequencePage(props) {
+  const iframeRef = useRef(null);
+
+  const handleResizeIframe = useCallback(() => {
+    // TODO: This won't work because of crossdomain issues.  Leaving here for reference once we're
```
Comment (davidjoy) [inline src/learning-sequence/LearningSequencePage.jsx:56]: Note this TODO.

File: src/learning-sequence/PageLoading.jsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
+import React, { Component } from 'react';
```
Comment (davidjoy) [inline src/learning-sequence/PageLoading.jsx:1]: I just snagged this component from a different MFE.  We should have one of these in Paragon.

File: .env.development (hunk 1 above)
Comment (abutterworth) [reply .env.development:2]: 🎈

File: src/index.jsx (hunk 2 above)
Comment (abutterworth) [reply src/index.jsx:23]: Good thinking

File: src/learning-sequence/LearningSequencePage.jsx [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+  const [data, setData] = useState(null);
+  const [loading, setLoading] = useState(false);
+  const [loaded, setLoaded] = useState(false);
+  const [failed, setFailed] = useState(false);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    setLoading(true);
+    apiFunction().then((response) => {
+      const result = format ? camelCaseObject(response.data) : response.data;
```
Comment (abutterworth) [inline src/learning-sequence/LearningSequencePage.jsx:23]: nice

**Review by abutterworth — APPROVED** [review]

**Pull request merged**